internal/core/propfilters/filter_test.go

```diff
@@ -0,0 +1,461 @@
+package propfilters
+
+import (
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFilters(t *testing.T) {
+	t.Run("Filter a single property name", func(t *testing.T) {
+		f, _ := New([]string{"pod-template-hash"}, []string{"*"},
+			[]string{"*"}, []string{"*"})
+
+		properties := map[string]string{"pod-template-hash": "123", "replicaSet": "abc"}
+		filteredProperties := f.FilterProperties(properties)
+
+		spew.Dump(f)
+		assert.True(t, len(filteredProperties) == 1)
```

Comment:
Use the specific assertion functions for length and equals instead of just using assert.True for everything.

Reply:
Also, assert.Equal works with maps I think so that you can just have one assertion and it will cover both length and content.